games/goradius/goradius/game.go

```diff
@@ -0,0 +1,297 @@
+package goradius
+
+import (
+	"embed"
+	"math/rand/v2"
+	"slices"
+
+	"github.com/sago35/koebiten"
+	"tinygo.org/x/drivers/pixel"
+)
+
+//go:embed *.png
+var fsys embed.FS
+
+const (
+	gameStateStart = iota
+	gameStatePlaying
+	gameStateGameOver
+
+	width  = 128
+	height = 64
+
+	gopherWidth  = 20
+	gopherHeight = 25
+
+	beamMax      = 1
+	beamCooldown = 60
+)
+
+type Game struct {
+	gopher            *koebiten.Image
+	x, y              int
+	scale             float32
+	theta             float32
+	score             int
+	beamEnergy        int
+	beamActive        bool // ビームが有効かどうかのフラグ
+	beamCooldownTimer int  // ビームのクールダウンタイマー
+
+	gameState int // ゲームの状態を管理する変数
+}
+
+type enemy struct {
+	enemyX int
+	enemyY int
+	speed  int // 敵の移動速度
+}
+
+var (
+	white = pixel.NewMonochrome(0xFF, 0xFF, 0xFF)
+	black = pixel.NewMonochrome(0x00, 0x00, 0x00)
+
+	x = float32(20.0)
+	y = float32(30.0)
+
+	frames   = 30         // フレーム数
+	interval = 120        // 敵の追加間隔
+	enemie   = []*enemy{} // 敵のX座標とY座標
+)
+
+func NewGame() *Game {
+	game := &Game{
+		gopher: koebiten.NewImageFromFS(fsys, "gopher.png"),
+		x:      width / 2,
+		y:      height / 2,
+		scale:  1,
+	}
+	return game
+}
+
+// Game update process
+func (g *Game) Update() error {
+	ds := float32(0.05)
+	dt := float32(0.2)
+	speed := 1
+	dx := 1 * speed
+	dy := 1 * speed
+
+	// スタート画面からゲームプレイ画面に遷移
+	if koebiten.IsKeyPressed(koebiten.KeyRotaryButton) {
+		g.gameState = gameStatePlaying
```

Comment:
Also, since I’d like it to work on as many other boards as possible, it would be helpful if the game could be played using just the `up`, `down`, `left`, `right`, `Key0`, and `Key1` buttons. The `KeyRotaryButton` should be set up as an additional option.